thirdparty/dns/JSDNS.jsm

```diff
@@ -487,17 +491,23 @@ function DNS_get_OS_DNSServers() {
 		var stream_filestream;
 		try {
 			/** @type {nsIFile} */
-			var resolvconf = Components.classes["@mozilla.org/file/local;1"].createInstance(Components.interfaces.nsIFile);
+			var resolvconf = Cc["@mozilla.org/file/local;1"] ? Cc["@mozilla.org/file/local;1"].createInstance(Ci.nsIFile) : null;
```

Comment:
Just as information: Note that this types of changes there not done because they could happen in practice. Main motivation was to make the TypeScript checks happy.
Still OK to back port them, even if it looks a lot more verbose than with the new optional chaining operator.

Reply:
Thank you. I removed the checks again for better readability